Automattic/jetpack
# Connection: offline mode for 127.0.0.1 and proper error messages

## Proposed changes:
* Add 127.0.0.1 into the list of known offline domain names to force offline mode upon it
* Display proper connection error message if trying to connect with a known local domain name

### Other information:


## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Load your monorepo site via `127.0.0.1`. Confirm Jetpack is in Offline Mode.
2. Comment out `127.0.0.1` [in the Status package](https://github.com/Automattic/jetpack/blob/0871825b0bad8f45ca966135e73855c1fc6bde85/projects/packages/status/src/class-status.php#L186).
3. Go to the Connection Screen and try connecting. You should see the error message: _"Your site host is on a private network. Jetpack can only connect to public sites."_
